python/ray/util/client/client_pickler.py

```diff
@@ -96,7 +96,7 @@ def persistent_id(self, obj):
         elif isinstance(obj, ClientRemoteFunc):
             if obj._ref is None:
                 obj._ensure_ref()
-            if type(obj._ref) == InProgressSentinel:
+            if type(obj._ref) is InProgressSentinel:
                 return PickleStub(
```

Comment:
Maybe we can use `isinstance(obj, target_instance)` here?
Its more concise, readable, and faster?

https://switowski.com/blog/type-vs-isinstance/

Reply:
@jjyao ? I am just fixing the lint..

`instance` will also allow sub class instances. I am not sure if it is safe for this case.